crazystocks/ARCHITECTURE.md

```diff
@@ -0,0 +1,105 @@
+# Goals
+
+The goal is to produce a small toolkit and library for evaluating stock market strategies based on predictive approaches.
+
+By predictive approaches, we mean machine learning on quantities related to the stock market. The quantities may include prices, volumes, expected gains/losses under some action.
+
+By stragegies, we mean simple rule based strategies which use the predictions of the machine learning models
+and may apply simple techniques as thresholding, stop-loss rules, time of day and so on.
+
+The same code should be able to run on:
+
+- the analyst's laptop using the DBG public dataset
+- the quantopian.com website
+
+# Stretch goals
+
+Ideally, we should be able to demonstrate consistent gains on the models/strategies across multiple stocks
```

Comment:
How does this improve upon the already large swaths of existing backtesting tools? I've written backtesters, I prefer to use existing tools. Writing a backtester in C++ was a nightmare to get right. Same with Python.

Here are some examples of battle tested backtesters:

**Python**

* [Zipline](https://www.zipline.io/)
* [Bt](http://pmorissette.github.io/bt/)
* [Backtrader](https://github.com/backtrader/backtrader)

**R**

* [Backtest](https://cran.r-project.org/web/packages/backtest/backtest.pdf)
* [Systematic Investor Toolbox](https://systematicinvestor.wordpress.com/systematic-investor-toolbox/)

Additionally machine learning can be shoehorned into most of these without much song and dance.

Reply:
Thanks! Will look into those if they can support what I need/want. May be they will.